lib/puppet/type/package.rb

```diff
@@ -437,12 +437,6 @@ def change_to_s(currentvalue, newvalue)
 
     newparam(:allow_virtual, :boolean => true, :parent => Puppet::Parameter::Boolean, :required_features => :virtual_packages) do
       desc 'Specifies if virtual package names are allowed for install and uninstall.'
-
-      # In a future release, this should be defaulted to true and the below deprecation warning removed
-      defaultto do
-        Puppet.deprecation_warning('The package type\'s allow_virtual parameter will be changing its default value from false to true in a future release. If you do not want to allow virtual packages, please explicitly set allow_virtual to false.') unless value
-        false
```

Comment:
The point of the deprecation warning was that it would eventually default to true so that rpm providers like yum can (in theory) behave like they do on the command line with respect to virtual packages.  If it defaults to false here, we added this (annoying) deprecation for no reason.  Removing the warning is definitely the correct thing to do, but I could go either way with respect to the default.